cmd/state-offline-installer/runInstall.go

```diff
@@ -0,0 +1,187 @@
+			}
+		}
+	}
+
+	if err := os.Mkdir(assetsPath, os.ModePerm); err != nil {
+		return errs.Wrap(err, "Unable to create assetsPath")
+	}
+
+	if err := os.Mkdir(artifactsPath, os.ModePerm); err != nil {
+		return errs.Wrap(err, "Unable to create artifactsPath directory")
+	}
+
+	err = unzip(params.backpackZipFile, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to unzip to assetsPath")
+	}
+
+	eulaAccepted, err := showEULAAndGetAcceptance(&prompter, licenseFilePath)
+	if err != nil {
+		return errs.Wrap(err, "Error with EULA acceptance")
+	}
+
+	if !eulaAccepted {
+		return locale.NewInputError("err_eula_not_accepted", "End user didn't accept the EULA")
+	}
+
+	out.Print(fmt.Sprintf("Stage 1 of 2 Start: Decompressing assets into: %s", assetsPath))
+	err = unzip(params.backpackZipFile, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Error unzipping assets")
+	}
+	out.Print(fmt.Sprintf("Stage 1 of 2 Finished: Decompressing assets into: %s", assetsPath))
+
+	archivePath := filepath.Join(assetsPath, artifactsTarGZName)
+
+	out.Print(fmt.Sprintf("Stage 2 of 2 Start: Decompressing artifacts into: %s", artifactsPath))
+	err = unpackArtifact(unarchiver.NewTarGz(), archivePath, artifactsPath)
+	if err != nil {
+		return errs.Wrap(err, "Could not unpack artifact %s", archivePath)
+	}
+	out.Print(fmt.Sprintf("Stage 2 of 2 Finished: Decompressing artifacts into: %s", artifactsPath))
+
+	offlineTarget := target.NewOfflineTarget(installToDir, artifactsPath)
+
+	// FIX: Need to do this more correctly with a New function
+	mockProgress := &OfflineProgressOutput{out: out}
+	// mockProgress := offlineProgressOutput.New(out)
+	logfile, err := buildlogfile.New(outputhelper.NewCatcher())
+	if err != nil {
+		return errs.Wrap(err, "Unable to create new logfile object")
+	}
+	eventHandler := events.NewRuntimeEventHandler(mockProgress, nil, logfile)
+
+	rti, err := runtime.New(offlineTarget, analytics, nil)
+	if err != nil && runtime.IsNeedsUpdateError(err) {
+		err = rti.Update(nil, eventHandler)
+		if err != nil {
+			return errs.Wrap(err, "Had an installation error")
+		}
+	}
+
+	configureEnvironmentAccepted, err := prompter.Confirm("Setup", "Setup environment for installed project?", &default_boolean_answer)
+	if err != nil {
+		return errs.Wrap(err, "Error getting confirmation")
+	}
+
+	if configureEnvironmentAccepted {
+		// This is the code to setup the environment when we install...
+		cfg, err := config.New()
+		if err != nil {
+			return errs.Wrap(err, "Error configuring environment")
+		}
+
+		env, err := rti.Env(false, false)
+		if err != nil {
+			return errs.Wrap(err, "Error setting environment")
+		}
+
+		sshell := subshell.New(cfg)
+
+		if rt.GOOS == "windows" {
+			contents, err := assets.ReadFileBytes("scripts/setenv.bat")
+			if err != nil {
+				return errs.Wrap(err, "Error reading file bytes")
+			}
+			err = fileutils.WriteFile(filepath.Join(installToDir, "setenv.bat"), contents)
```

Comment:
Note: better not to reuse the err variable as it can break nil type assertions when two different errors pass a different type (both of which will satisfy the error interface).